### ISSUE TITLE
`rucio --help` fails on misconfigured sites

### ISSUE THREAD

**Issue by bbockelm** [rucio/rucio#1911]
Motivation
----------
Even on a bad installation or typos in the configuration system, `rucio --help` ought to work.

The current version of `rucio` imports the `DownloadClient` at the module level.  If there is an error in the `rucio.cfg` that causes the `DownloadClient` constructor to throw an exception, you can't get to `--help`.

Example:
```
$ rucio --help 
Traceback (most recent call last):
  File "/home/cse496/bbockelm/software/rucio-doma/bin/rucio", line 63, in <module>
    from rucio.client.downloadclient import DownloadClient
  File "/home/cse496/bbockelm/projects/rucio/lib/rucio/client/downloadclient.py", line 34, in <module>
    client = Client()
... stack trace goes on for miles ...
  File "/home/cse496/bbockelm/software/rucio-doma/lib/python2.7/site-packages/requests/adapters.py", line 412, in send
    self.cert_verify(conn, request.url, verify, cert)
  File "/home/cse496/bbockelm/software/rucio-doma/lib/python2.7/site-packages/requests/adapters.py", line 227, in cert_verify
    "invalid path: {0}".format(cert_loc))
IOError: Could not find a suitable TLS CA certificate bundle, invalid path: /etc/ca.crt
```

In my case, the problem was caused by an invalid path in `rucio.cfg` for the CA bundle.

Modification
------------
Failures to create the clients should be swallowed until they are actually used.  Potentially we shouldn't even import the clients until needed.

**Comment by bbockelm** [comment]
Note that importing the client at the top of the `rucio` CLI causes an authentication to be attempted even when you're just doing `rucio --help`.

**Comment by TomasJavurek** [comment]
That doesn't seem to be problem only with importing downloadclient but Clients in genereal. The config is imported in baseclient.

**Comment by bari12** [comment]
Just importing the module should not read the config? Only when constructing the client object; Then indeed it does read the config and even trys to authenticate against the rucio server.
@TomasJavurek can you please check the code path why this is the case even when doing `rucio --help`? In my opinion no client object needs to be constructed in that case.

**Comment by rcarpa** [comment]
The particular issue reported here was probably fixed in 5990e5d56f822227eeb6b7cd7d779e5bfdc06098 . I cannot reproduce it anymore. 
There is however a case when the config file is completely missing: #4601
